### PR TITLE
Add HotRod no downtime store for Realms

### LINKED ISSUE #9670: Hot Rod map storage: Realm no-downtime store
### Description

Create a Hot Rod map storage implementation for Realm area

### PR DESCRIPTION
**Pull request by mhajas**
Closes #9670

### CODE REVIEW

**Review by martin-kanis — CHANGES_REQUESTED** [review]
Thanks @mhajas! The PR looks good. Could you just reindex `realms` cache in testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json?

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/realm/HotRodRealmEntity.java [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+@GenerateHotRodEntityImplementation(
+        implementInterface = "org.keycloak.models.map.realm.MapRealmEntity",
+        inherits = "org.keycloak.models.map.storage.hotRod.realm.HotRodRealmEntity.AbstractHotRodRealmEntityDelegate"
+)
+@ProtoDoc("@Indexed")
+public class HotRodRealmEntity extends AbstractHotRodEntity {
+    @ProtoField(number = 1, required = true)
+    public int entityVersion = 1;
+
+    @ProtoDoc("@Field(index = Index.YES, store = Store.YES)")
```
[the remark this answers is not in the document]
Comment (mhajas) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/realm/HotRodRealmEntity.java:59]: Looking at the `SearchableFields` for the `RealmModel`, I would say we should leave the index as is. Currently, we don't use `ID` in any MCB, however, we cannot ensure it won't happen in the future version. WDYT?

**Comment by mhajas** [comment]
Thanks for the review @martin-kanis, added&rebased

File: model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/realm/HotRodRealmEntity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhajas) [reply model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/realm/HotRodRealmEntity.java:59]: Let's have it without the index now. At least we can try the index migration if we decide to add MCB with ID later.

**Comment by martin-kanis** [comment]
@mhajas Can you please check failing `RoleModelTest` with `map-ldap` profile?

**Comment by mhajas** [comment]
@martin-kanis the failures in the model tests are caused by this issue: https://github.com/keycloak/keycloak/issues/11171

**Review by martin-kanis — APPROVED** [review]
Thanks @mhajas. The failure in CrossDC tests doesn't seem related. Here is similar failure from `main` https://github.com/keycloak/keycloak/runs/5823731955?check_suite_focus=true

**Review by hmlnarik — APPROVED** [review]
Approving per @martin-kanis 's approval

**Pull request merged**